src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcEndpointDataSource.cs

```diff
@@ -392,18 +398,7 @@ private IChangeToken GetCompositeChangeToken()
             return new CompositeChangeToken(changeTokens);
         }
 
-        public override IChangeToken ChangeToken
-        {
-            get
-            {
-                if (_changeToken == null)
-                {
-                    _changeToken = GetCompositeChangeToken();
-                }
-
-                return _changeToken;
-            }
-        }
+        public override IChangeToken ChangeToken => GetCompositeChangeToken();
```

Comment:
Should the ChangeToken property instead be a GetChangeToken method? I get the feeling looking at everywhere IChangeToken it is used that a fresh change token should be returned, and a method is more appropriate.

Reply:
https://github.com/aspnet/Routing/issues/634